tests/test_helpers/test_env_vars.py

```diff
@@ -195,3 +197,20 @@ def test_match_export_parse_consistency(export_data, template) -> None:
     parsed_data = apply_env_variables(template, env_variables=env_vars)
 
     assert parsed_data == export_data
+
+
+def test_apply_env_variables_on_agent_config():
+    """Test apply_env_variables_on_agent_config function."""
+    result = apply_env_variables_on_agent_config(
+        [{"arg": "${VAR}"}, {"public_id": "fetchai/test:0.1.0", "arg": "${VAR}"}],
+        {"VAR": 12},
+    )
+    assert result == [{"arg": 12}, {"arg": 12, "public_id": "fetchai/test:0.1.0"}]
+
+
+def test_is_strict_list():
+    """Test is_strict method."""
+    assert is_strict_list([1, 2, 3])
+    assert is_strict_list([1, 2, 3, [1, 2, 3]])
+    assert not is_strict_list([1, 2, {}])
+    assert not is_strict_list([1, 2, [[{}]]])
```

Comment:
the original function is not clear and should likely not exist. 

is_strict_list([tuple(dict(hello="world"), )])
>>> True

Reply:
good point.
for me, it looks like a check for nested dicts

added support of tuple

```assert is_strict_list([tuple(dict(hello="world"), )])``` - is true, tuple(dict) => tuple of dict keys

Reply:
```
def is_container(obj: Any) -> bool:
    return isinstance(obj, (MutableSequence, MutableSet, MutableMapping))

def is_strict_list(data: List):
    for c in filter(is_container, data):
        if isinstance(c, list):
            return is_strict_list(c)
        return False
    return True
```

Reply:
good point and nice to have a ticket with proposals.